src/posix/platform/infra_if.cpp

```diff
@@ -515,6 +524,81 @@ void InfraNetif::ReceiveIcmp6Message(void)
     }
 }
 
+void InfraNetif::HandleHostAddressResponse(union sigval sv)
+{
+    otError          error = OT_ERROR_NONE;
+    struct gaicb *   req   = (struct gaicb *)sv.sival_ptr;
+    struct addrinfo *res   = (struct addrinfo *)req->ar_result;
+
+    const uint8_t kMaxHostAddresses = 10;
+    otIp6Address  addresses[kMaxHostAddresses];
+    uint8_t       numAddresses = 0;
+
+    otLogInfoPlat("Handling host address response...");
+
+    for (struct addrinfo *rp = res; rp != NULL && numAddresses < kMaxHostAddresses; rp = rp->ai_next)
+    {
+        struct sockaddr_in6 *ip6Addr;
+
+        if (rp->ai_family != AF_INET6)
+        {
+            continue;
+        }
+
+        ip6Addr = reinterpret_cast<sockaddr_in6 *>(rp->ai_addr);
+        memset(&addresses[numAddresses], 0, sizeof(otIp6Address));
+        memcpy(&addresses[numAddresses].mFields.m8, &ip6Addr->sin6_addr.s6_addr, OT_IP6_ADDRESS_SIZE);
```

Comment:
```suggestion
        memcpy(&addresses[numAddresses].mFields.m8, &ip6Addr->sin6_addr.s6_addr, OT_IP6_ADDRESS_SIZE);
```
if we are memcpying the full OT_IP6_ADDRESS_SIZE size, is memset still necessary?